src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -174,6 +174,8 @@ private static ImmutableArray<Diagnostic> EmitRegexMethod(IndentedTextWriter wri
                 return ImmutableArray.Create(Diagnostic.Create(DiagnosticDescriptors.LimitedSourceGeneration, rm.MethodSyntax.GetLocation()));
             }
 
+            bool allowUnsafe = compilation.Options is CSharpCompilationOptions { AllowUnsafe: true };
```

Comment:
if all we use from compilation is this check, is it better to just pass the bool from the generator or do you expect us to use the Compilation object for something else?

Reply:
As long as we're passing something, seemed like we might as well pass around the Compilation in case we need anything more from it later.